### PR TITLE
[protobuf] Fix protoc.exe use the new headers with old DLLs

### PR DESCRIPTION
**Pull request by FrankXie05**
Fix [vcpkg testing MSVC](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=8132201&view=results) issues.
```
cmd.exe /C "cd /D D:\buildtrees\gamenetworkingsockets\x64-windows-dbg\src && D:\installed\x64-windows\tools\protobuf\protoc.exe 
--cpp_out D:/buildtrees/gamenetworkingsockets/x64-windows-dbg/src 
-I D:/buildtrees/gamenetworkingsockets/src/687597c3e9-05f1ecc051.clean/src/common D:/buildtrees/gamenetworkingsockets/src/687597c3e9-05f1ecc051.clean/src/common/steamnetworkingsockets_messages_certs.proto"
ninja: build stopped: subcommand failed.
```

According to the suggestion of Stephan, add compiler option `/D_DISABLE_CONSTEXPR_MUTEX_CONSTRUCTOR` to workaround this issue.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by Neumann-A** [comment]
I don't think this is the right solution. Please explain the problem further.

**Comment by FrankXie05** [comment]
@BillyONeal Should we show information about our internal testing to users?

**Comment by Osyotr** [comment]
What new headers? What old DLLs? What does adding this definition fix?

File: ports/protobuf/portfile.cmake [hunk 1]
```diff
@@ -11,6 +11,12 @@ vcpkg_from_github(
         fix-default-proto-file-path.patch
         compile_options.patch
 )
+if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "19.38.32920.95")
```
Comment (Neumann-A) [inline ports/protobuf/portfile.cmake:15]: this is not available in portfile.cmake since script mode does not run compiler detection

Comment (FrankXie05) [reply ports/protobuf/portfile.cmake:15]: Shuold be added `CMakeLists.txt`.

**Review by BillyONeal — CHANGES_REQUESTED** [review]
I don't think we should be working around an MSVC++ compiler team infrastructure issue by changing downstream consumers.

**Comment by Cheney-W** [comment]
Does this mean we need to keep these failures in the result of `VCPKG Testing MSVC run` until the `MSVC++ compiler team infrastructure issue` is fixed?

**Comment by BillyONeal** [comment]
> Does this mean we need to keep these failures in the result of `VCPKG Testing MSVC run` until the `MSVC++ compiler team infrastructure issue` is fixed?

Yes. Compiler team needs to fix the infrastructure for their tests.

**Comment by FrankXie05** [comment]
Close this PR due to https://devdiv.visualstudio.com/DevDiv/_git/msvc/pullrequest/494762.

At present, this problem is avoided by modifying the code logic and macro `_ENABLE_CONSTEXPR_MUTEX_CONSTRUCTOR`.  
Unless we actively pass the macro.

And, this behavior will be the default in 17.9 and this issue will be completely resolved.

**Pull request closed without merging**